src/classes/PS_CommitmentRequest.cls

```diff
@@ -165,7 +193,15 @@ public inherited sharing class PS_CommitmentRequest {
             .withProductMetaData(productMetaData);
 
         setDonorInfo(reqBody, rd);
-        setSchedules(reqBody, rd.RecurringDonationSchedules__r, currencyCode);
+        setSchedules(reqBody, scheduleService.buildNewSchedules(rd), currencyCode);
```

Comment:
Need to build schedules for both create and edit commitment since RD record is not inserted/updated in CRM yet. (W-8554656)